### PR TITLE
commands: rewrite add/deltime to use common code

### PR DESCRIPTION
**Pull request by sit**
Extract common code between addtime and deltime into functions.
Combine them into a single source file (adjusttime.js).
Use userRepository from persistence.js instead of load/saveUserData.

### CODE REVIEW

File: commands/admin/adjusttime.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    return adjustTime(message, userIdArg, shouldNegate ? -delta : delta)
+  }
+
+  async function adjustTime (message, userId, delta) {
+    let user = await client.userRepository.load(userId)
+    if (user === null) {
+      return client.errorMessage(message, `User <@${userId}> doesn't exist in DB`)
+    }
+
+    user.current_session_playtime = Math.max(0, user.current_session_playtime + delta)
```
Comment (sit) [inline commands/admin/adjusttime.js]: This also happens to prevent negative numbers passed to `addtime` from messing anyone up.

**Comment by sit** [comment]
@pianiststickman

File: commands/admin/adjusttime.js (hunk 1 above)
Comment (pianiststickman) [reply commands/admin/adjusttime.js]: Maybe we should also reject delta <= 0. I can't see why anybody would ever pass a non-positive delta unless they're trying to be clever.

File: commands/admin/adjusttime.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    user.overall_session_playtime = Math.max(0, user.overall_session_playtime + delta)
+
+    return client.userRepository.save(user).then(res => {
```
Comment (pianiststickman) [inline commands/admin/adjusttime.js]: Is it not possible to try/catch an `await` on the call to `save()` here? Just seems a little odd to call `.then()` if we're trying to standardize our usage of async/await.

File: commands/admin/adjusttime.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
+module.exports.load = (client) => {
```
Comment (sit) [inline commands/admin/adjusttime.js]: @pianiststickman I was trying to be clever here and reduce the duplication between `addtime` and `deltime`, whose implementations are basically the same.

**Pull request closed without merging**